lib/TH/generic/THVectorDispatch.c

```diff
@@ -0,0 +1,99 @@
+#ifndef TH_GENERIC_FILE
+#define TH_GENERIC_FILE "generic/THVectorDispatch.c"
+#else
+
+#include "simd/simd.h"
+#include <assert.h>
+
+/* For now there are only SIMD implementations for FLOAT and DOUBLE.
+ * Hopefully in the future this can be made totally generic (e.g, there are SIMD implementations
+ * for a lot of functions */
+/* Each function with multiple implementations has:
+ * 1. A DISPATCHPTR which will be initialized to point to the best available implementation for the host
+ * 2. A DISPATCHTABLE which holds pointers to each implementation of a function, and a value indicating
+ *    which SIMD extension a given implementation uses
+ * 3. A dispatch stub, which is what is actually called by clients, that simply wraps the dispatch pointer.
+ */
+
+static void (*THVector_(fill_DISPATCHPTR))(real *, const real, const long) = &THVector_(fill_SCALAR);
+static FunctionDescription THVector_(fill_DISPATCHTABLE)[] = {
+#if defined(TH_REAL_IS_DOUBLE) || defined(TH_REAL_IS_FLOAT)
+  FUNCTION_IMPL(THVector_(fill_SSE), SIMDExtension_SSE),
+#endif
+  FUNCTION_IMPL(THVector_(fill_SCALAR), SIMDExtension_SCALAR)
+};
+void THVector_(fill)(real *x, const real c, const long n) {
+  assert(THVector_(fill_DISPATCHPTR));
```

Comment:
Are these asserts really needed? You explicitly initialize the pointers a couple lines above. I think it makes sense to do any of the two (with 1. being a bit better I think):
1. Don't initialize the pointer explicitly, leave the assert - this allows to catch errors if someone forgets to add `INIT_DISPATCH_PTR` below.
2. Initialize the pointer explicitly, remove the assert - it won't be triggered anyway.